integrade/tests/conftest.py

```diff
@@ -60,3 +67,40 @@ def instances_to_terminate():
         with Pool(num_instances) as p:
             p.map(
                 terminate_instance, instances_to_terminate)
+
+
+@pytest.fixture()
+def cloudtrails_to_delete():
+    """Provide list to test to indicate cloudtrails that should be terminated.
+
+    We must know what aws profile to use, so append tuples of (aws_profile,
+    cloudtrail_name) to the list.
+    """
+    cloudtrails_to_delete = []
+
+    yield cloudtrails_to_delete
+
+    if cloudtrails_to_delete:
+        num_trails = len(cloudtrails_to_delete)
+        with Pool(num_trails) as p:
+            p.map(
+                delete_cloudtrail, cloudtrails_to_delete)
+
+
+@pytest.fixture()
+def cloudtrails_and_buckets_to_delete():
+    """Provide list to test to indicate cloudtrails and buckets to delete.
+
+    We must know what aws profile to use, so append tuples of (aws_profile,
+    cloudtrail_name, s3_bucket_name) to the list.
+    """
+    to_delete = []
+
+    yield to_delete
+
+    if to_delete:
+
+        num_trails = len(to_delete)
+        with Pool(num_trails) as p:
```

Comment:
Usually the Pool size should be the number of available cores, supposing that we have many trails in the future like 10 and we have only 2 CPUs we are going to create more collision and interrupts than being fast as possible.

That said if you don't pass the number of process to have on the Pool then the cpu_count will be used:

> processes is the number of worker processes to use. If processes is None then the number returned by os.cpu_count() is used.

Reply:
ahhh. Ok so if there are 15 things to map the function to, and I have 4 cores, it will do batches of 4 and then join at the end?

Reply:
Yes, check this out:

```
In [1]: import multiprocessing

In [2]: import random

In [4]: import time

In [7]: def long_processing(timeout):
   ...:     print(f'going to sleep for {timeout}')
   ...:     time.sleep(timeout)
   ...:     print(f'done sleeping for {timeout}')
   ...:     

In [8]: with multiprocessing.Pool() as pool:
   ...:     pool.map(long_processing, [random.randint(1,5) for _ in range(10)])
   ...:     
going to sleep for 2
going to sleep for 3
going to sleep for 3
going to sleep for 4
done sleeping for 2
going to sleep for 2
done sleeping for 3
done sleeping for 3
going to sleep for 5
going to sleep for 4
done sleeping for 2
done sleeping for 4
going to sleep for 2
going to sleep for 1
done sleeping for 1
going to sleep for 3
done sleeping for 2
done sleeping for 4
done sleeping for 5
done sleeping for 3

In [10]: multiprocessing.cpu_count()
Out[10]: 4
```

It will get 4 to process in the beginning and will be assigning new as needed.

Reply:
cooool